examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,243 @@
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+	log.Print("Starting Health Ping")
+	go doHealth(s, sigCtx)
+
+	log.Print("Marking this server as ready")
+	if err := s.Ready(); err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+
+	if *simEndpoint != "" {
+		log.Printf("Creating Sim Client at endpoint %s", *simEndpoint)
+		simConn = initClient(*simEndpoint, *simContext, "Sim")
+	}
+
+	if *genAiEndpoint == "" {
+		log.Fatalf("GenAiEndpoint must be specified")
+	}
+	log.Printf("Creating GenAI Client at endpoint %s", *genAiEndpoint)
+	genAiConn = initClient(*genAiEndpoint, *genAiContext, "GenAI")
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(*port, genAiConn)
+	} else {
+		log.Printf("Starting autonomous chat with Prompt: %s", *prompt)
+		go autonomousChat(*prompt, genAiConn, simConn, *numChats)
+	}
+
+	<-sigCtx.Done()
+	os.Exit(0)
+}
+
+func initClient(endpoint string, context string, name string) *connection {
+	// TODO: create option for a client certificate
+	client := &http.Client{}
+	conn := &connection{client: client, endpoint: endpoint, context: context, name: name}
+	return conn
+}
+
+type connection struct {
+	client   *http.Client
+	endpoint string // full base URL for API requests
+	context  string
+	name     string // human readable name for the connection
+	// TODO: create options for routes off the base URL
+}
+
+type GenAIRequest struct {
+	Context string `json:"context,omitempty"`
+	Prompt  string `json:"prompt"`
+}
+
+func handleGenAIRequest(prompt string, clientConn *connection) (string, error) {
+	jsonRequest := GenAIRequest{
+		Context: clientConn.context,
+		Prompt:  prompt,
+	}
+	jsonStr, err := json.Marshal(jsonRequest)
+	if err != nil {
+		return "", fmt.Errorf("unable to marshal json request: %v", err)
+	}
+
+	req, err := http.NewRequest("POST", clientConn.endpoint, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		return "", fmt.Errorf("unable create http POST request: %v", err)
+	}
+	req.Header.Set("accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := clientConn.client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("unable to post request: %v", err)
+	}
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("unable to read response body: %v", err)
+	}
+	defer resp.Body.Close()
+	body := string(responseBody)
+
+	if resp.StatusCode != 200 {
+		err = fmt.Errorf("Status: %s, Body: %s", resp.Status, body)
+	}
+	return string(responseBody) + "\n", err
+}
+
+// Two AIs (connection endpoints) talking to each other
+func autonomousChat(prompt string, conn1 *connection, conn2 *connection, numChats int) {
+	if numChats <= 0 {
+		return
+	}
+	response, err := handleGenAIRequest(prompt, conn1)
+	if err != nil {
+		log.Fatalf("could not send request: %v", err)
+	} else {
```

Comment:
Since the if clause causes the program to exit, this doesn't need to be in an else clause and can be un-indented, e.g.

```
if err != nil {
	log.Fatalf("could not send request: %v", err)
}
log.Printf("...")
```